converter.go

```diff
@@ -1736,7 +1949,11 @@ func buildArrowListRecursive[T any](ctx context.Context, rowIdx int, fieldMetada
 		if err != nil {
 			return nil, err
 		}
+		if arrowList == nil {
```

Comment:
Is this variable untyped? Is this condition tested in tests?

Reply:
There is for sure a scenario where `buildListFromNativeArrow` returns `nil, nil`, so this would panic if we didn't catch that. Lack of this condition probably means that this wasn't tested, I can try and create a case where this comes up

Reply:
Upon further inspection, this comes up in an already tested case. This is similar to all of the above discussions - nil can't be asserted to be of a certain type